addons/website/controllers/main.py

```diff
@@ -302,17 +297,14 @@ def attach(self, func, upload=None, url=None, disable_optimization=None):
                     if not disable_optimization and image.format in ('PNG', 'JPEG'):
                         image_data = image_save_for_web(image)
 
-                    attachment_id = Attachments.create(request.cr, request.uid, {
+                    attachment = Attachments.create({
                         'name': c_file.filename,
                         'datas': image_data.encode('base64'),
                         'datas_fname': c_file.filename,
                         'res_model': 'ir.ui.view',
-                    }, request.context)
-                    attachment_ids.append(attachment_id)
+                    })
+                    uploads.append({'id':attachment.id, 'website_url':attachment.website_url})
 
-                uploads = Attachments.read(
-                    request.cr, request.uid, attachment_ids, ['website_url'],
-                    context=request.context)
```

Comment:
This is inefficient: you will perform one request per attachment to read `website_url`. Simply make a list of attachments, and compute `uploads` from it:

    attachments = []
    for c_file in request.httprequest.files.getlist('upload'):
        ...
            attachments.append(Attachments.create({
                'name': c_file.filename,
                'datas': image_data.encode('base64'),
                'datas_fname': c_file.filename,
                'res_model': 'ir.ui.view',
            })
    uploads = [{'id':attachment.id, 'website_url':attachment.website_url} for attachment in attachments]